app/javascript/components/upload/UploadWizard.jsx

```diff
@@ -269,49 +270,72 @@ export function RawUploadWizard({ studyAccession, name }) {
 
   /** save the given file and perform an upload if a selected file is present */
   async function saveFile(file) {
+    let fileToSave = file
     let studyFileId = file._id
 
+    // in AnnData case of saving file when you're doing an update
     if (isAnnDataExperience) {
```

Comment:
For maintainability, could you extract lines 278 - 314 into a separate function?

Those 36 lines add substantially complexity to this `saveFile` function that's already likely a bit too complex.  Having the `isAnnDataExperience` logic in its own function would enable the containing `saveFile` function to be understand more easily at a high level, while also making it more testable (which would help robustness).